File: testdata/jennies/builders/struct_fields_as_args_assignment/PHPBuilder/src/Sandbox/SomeStructBuilder.php
<?php

namespace Grafana\Foundation\Sandbox;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\Sandbox\SomeStruct>
 */
class SomeStructBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\Sandbox\SomeStruct $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\Sandbox\SomeStruct();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\Sandbox\SomeStruct
     */
    public function build()
    {
        return $this->internal;
    }

    public function time(string $from, string $to): static
    {    
        if ($this->internal->time === null) {
            $this->internal->time = "unknown";
        }
        
        $this->internal->time->from = $from;
        $this->internal->time->to = $to;
    
        return $this;
    }

}

File: testdata/jennies/builders/struct_fields_as_args_assignment/JavaBuilder/sandbox/SomeStructBuilder.java
package sandbox;


public class SomeStructBuilder implements cog.Builder<SomeStruct> {
    protected final SomeStruct internal;
    
    public SomeStructBuilder() {
        this.internal = new SomeStruct();
    }
    public SomeStructBuilder time(String from,String to) {
		if (this.internal.time == null) {
			this.internal.time = new Object();
		}
        this.internal.time.from = from;
        this.internal.time.to = to;
        return this;
    }
    public SomeStruct build() {
        return this.internal;
    }
}
